ocs_ci/helpers/mcg_stress_helper.py

```diff
@@ -0,0 +1,618 @@
+                    f"METADATA OP: Successfully completed noobaa account creation/update/deletion operation"
+                    f" in the background for the iteration {current_iteration}"
+                )
+                break
+
+    # run the above metadata intense ops parallel
+    logger.info("Initiating metadata ops")
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+    futures_obj = list()
+    futures_obj.append(executor.submit(_run_bucket_ops))
+    futures_obj.append(executor.submit(_run_object_metadata_ops))
+    futures_obj.append(executor.submit(_run_noobaa_account_ops))
+    logger.info("Waiting until all metadata operations are completed")
+    for future in as_completed(futures_obj):
+        future.result()
+    executor.shutdown()
+
+
+def delete_objs_from_bucket(pod_obj, bucket, prev_iteration, event=None, multiplier=1):
+    """
+    Delete all the objects from a bucket
+
+    Args:
+        pod_obj (Pod): Noobaa stress CLI pod object
+        bucket (Tuple): Tuple consisting of backend storage type and bucket object
+        prev_iteration (int): Iteration number or prefix from where should delete objects
+        event (threading.Event()): Event object to signal the execution
+            completion
+
+    """
+    current_iteration = prev_iteration + 1
+    bucket_type, bucket_obj = bucket
+    bucket_name = bucket_obj.name
+    base_timeout = 20000
+    timeout = base_timeout * multiplier
+
+    if bucket_type.upper() == "RGW":
+        mcg_obj = OBC(bucket_name)
+    else:
+        mcg_obj = MCG()
+
+    logger.info(
+        f"DELETE OP: Delete objects recursively from the bucket "
+        f"{bucket_name} under prefix {prev_iteration}"
+    )
+
+    rm_object_recursive(
+        pod_obj,
+        bucket_name,
+        mcg_obj,
+        prefix=prev_iteration,
+        timeout=timeout,
+    )
+    logger.info(
+        f"DELETE OP: Total of {(multiplier-1)*1000000} objects got deleted "
+        f"in the current iteration {current_iteration}"
+    )
+    logger.info(
+        f"DELETE OP: Successfully completed object deletion operation on "
+        f"bucket {bucket_name} under prefix {prev_iteration}"
+    )
+
+
+def list_objs_from_bucket(bucket, prev_iteration, event=None):
+    """
+    List objects from bucket
+
+    Args:
+        bucket (Tuple): Tuple consisting of backend storage type and bucket object
+        prev_iteration (int): Iteration number or prefix from where should list objects
+        event (threading.Event()): Event object to signal the execution
+            completion
+
+    """
+    current_iteration = prev_iteration + 1
+    bucket_type, bucket_obj = bucket
+    bucket_name = bucket_obj.name
+    if bucket_type.upper() == "RGW":
+        mcg_obj = OBC(bucket_name)
+    else:
+        mcg_obj = MCG()
+
+    logger.info(
+        f"LIST OP: Listing objects from the bucket {bucket_name} "
+        f"under prefix {prev_iteration}"
+    )
+    while True:
```

Comment:
you can even exit if the event is not set 
while not event.is_set():

Reply:
@PrasadDesala Im repeating the list if the previous list is already completed. hence sustaining the stress